tests/test_ticks.c

```diff
@@ -64,53 +58,52 @@ static uint32_t test_cases[][3] = {
 	{0x2CB4178, 0x59682F0, 0x2CB4178},
 	// Wrap towards the end
 	{0xFD34BE87, 0x2CB4178, 0x59682F0},
-	// wait a long time - 1 second in the range to 1 second before the end
+	// Wait a long time - 1 second into the range => 1 second before the end
 	{0x2CB4178, 0xFD34BE87, 0xFA697D0F},
-	 // already fulfilled way past the start tick
-	{0x2CB4178, 0xFD34BE87, 0x59682F0},
-	// TODO: also test unfulfilled cases to make sure we don't exit the loop early.
+	// Long wait, wrapping around - 1 second before the end => 2 second before the end
+	{0xFD34BE87, 0xFA697D0F, 0xFD34BE87},
 };
 
 static uint32_t test_cases_ms[][3] = {
 	// No overflow
 	{0x2CB4178, 0x59682F0, 1000},
 	// Wrap towards the end
 	{0xFD34BE87, 0x2CB4178, 2000},
-	// wait a long time - 1 second in the range to 1 second before the end
-	{0x2CB4178, 0xFD34BE87, 88000},
-	 // already fulfilled way past the start tick
-	{0x2CB4178, 0xFD34BE87, 2000},
-	// TODO: also test unfulfilled cases to make sure we don't exit the loop early.
+	// Wait a long time - 1 second into the range => 1 second before the end
+	{0x2CB4178, 0xFD34BE87, 89626},
+	// Long wait, wrapping around - 1 second before the end => 2 second before the end
+	{0xFD34BE87, 0xFA697D0F, 90626},
 };
 
 void test_ticks(TestContext *ctx) {
 	uint32_t ticks_0;
 	uint32_t ticks_1;
 
-	uint32_t continue_ticks;
-	READ_TICKS(continue_ticks);
+	uint32_t continue_ticks = TICKS_READ();
 
 	disable_interrupts();
 
 	// Put the instructions on the same cacheline
-	asm volatile(".align 4");
-	WRITE_TICKS(0x0);
-	READ_TICKS(ticks_0);
-	WRITE_TICKS(0xFFFFFFFF);
-	READ_TICKS(ticks_1);
+	for (int i = 0; i < 2; i++) {
```

Comment:
The comment needs updating here. You may want to explain that you're looping twice to make sure everything is in icache.